go/private/rules/binary.bzl

```
@@ -130,10 +132,27 @@ def _go_binary_impl(ctx):
         goarch = ctx.attr.goarch,
     )
 
+    generated_srcs = []
+    data = ctx.attr.data
+    deps = ctx.attr.deps
+
+    if go.coverage_enabled and go.coverage_instrumented:
```

```suggestion
    if go.coverage_enabled:
```
The binary itself may not be covered by the filter, but have transitive deps that do

---

yep, good point. Also another (preexisting) issue - we only propagate for .go files across data, so binaries calling across languages also won't propagate coverage info. We may want to expand that to other files, perhaps in a separate change

---

What is it that we don't propagate? `instrumented_files_info` adds `data` as a `dependency_attribute`.

---

As far as I remember `extensions` only matters for the source attributes, instrumented files from dep attributes are always forwarded. Cross-language coverage should work with this.

---

oh ok, perfect